File: 6. Zigzag Conversion/zigzag_conversion.py
def zigzag (text, rows):
	out = []
	
	for a in range(rows):
		out.append("")
		
	len_text = len(text)
	
	straight = True
	text_cursor = 0
	
	while (text_cursor < len_text):
		if straight:
			for row_cursor in range(len(out)):
				if (text_cursor < len_text):
					out[row_cursor] += text[text_cursor]
					text_cursor += 1
				else:
					break
			straight = False
		else:
			for terminals in range(len(out) - 2):
				for row_cursor in range(len(out)):
					if (text_cursor < len_text):
						#if (row_cursor == (rows - 1 - (a % rows))): #
						if (row_cursor == (rows - 2 - terminals)):
							out[row_cursor] += text[text_cursor]
							text_cursor += 1
						else:
							out[row_cursor] += " "
					else:
						break
			straight = True
	
	return out
	
lines = zigzag("PAYPALISHIRING", 4)

for line in lines:
	print(line)

File: 6. Zigzag Conversion/zigzag_conversion.swift
func zigzag (text: String, rows: Int) -> Array<String> {
	var out: Array<String> = [];
	
	for a in 0..<rows{
		out.append("");
	}
	
	var len_text: Int = text.count;
	
	var straight: Bool = true;
	var text_cursor: Int = 0;
	
	while text_cursor < len_text{
		if (straight){
			for row_cursor in 0..<rows{
				if text_cursor < len_text{
					let index = text.index(text.startIndex, offsetBy: text_cursor);
					out[row_cursor].append(text[index]);
					text_cursor += 1;
				} else {
					break;
				}
			}
			straight = false;
		} else {
			for term in 0..<(rows - 2){
				for row_cursor in 0..<rows{
					if (text_cursor < len_text){
						if row_cursor == (rows - 2 - term){
							let index = text.index(text.startIndex, offsetBy: text_cursor);
							out[row_cursor].append(text[index]);
							text_cursor += 1;
						} else {
							out[row_cursor].append(" ");
						}
					} else {
						break;
					}
				}
			}
			straight = true;
		}
	}
	return (out);
}

	
let lines: Array<String> = zigzag(text: "PAYPALISHIRING", rows: 4);

print("This is the handy work of Swift\n\n\n");

for line in lines{
	print(line)
}

print("\n\n\n");
